### PR TITLE
create local network gateway resource

### PR DESCRIPTION
**Pull request by azgcloudev**
## Problem

Solving Issue #396 

## Solution
Adding resource

## Changes Made
added local network gateway
also created a function to get the resource groups in the subscription to be able to call all the local network gateways, as function do not call under a subscription level

## How to Test

Added a new local network gateway in the Azure subscription, test if it shows successfully in Komiser. Cost is not showing, as LNGs do not generate cost.

## Screenshots
image

## Notes

[Any additional notes or information that you would like to share with the reviewers.]

## Checklist

- [x] Code follows the <a href="https://github.com/tailwarden/komiser/blob/master/CONTRIBUTING.md">contributing</a> guidelines
- [x] Changes have been thoroughly tested
- [ ] <a href="https://github.com/tailwarden/docs.komiser.io">Documentation</a> has been updated, if necessary
- [ ] Any dependencies have been added to the project, if necessary

## Reviewers

@jakepage91

### CODE REVIEW

File: providers/azure/resourcegroup/resource_groups.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	pager := resourceGroupClient.NewListPager(nil)
+	for pager.More() {
+		page, err := pager.NextPage(ctx)
+		if err != nil {
+			return resourceGroupNames, err
+		}
+
+		for _, resourceGroup := range page.ResourceGroupListResult.Value {
+
+			resourceGroupNames = append(resourceGroupNames, *resourceGroup.Name)
```
Comment (AvineshTripathi) [inline providers/azure/resourcegroup/resource_groups.go:33]: can you also add other details like https://github.com/tailwarden/komiser/pull/1457/files#diff-e34e11157def2d0cf86866b6de46e377bda2def7d4aa338041dac225c5a1e7cfR56

File: providers/azure/networking/local_network_gateways.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+					Provider:   "Azure",
+					Account:    client.Name,
+					Service:    "Local Network Gateway",
+					Region:     *lng.Location,
+					ResourceId: *lng.ID,
+					Cost:       0,
+					Name:       *lng.Name,
+					FetchedAt:  time.Now(),
+					Tags:       tags,
+					Link:       fmt.Sprint("https://portal.azure.com/#resource%s", *lng.ID),
```
Comment (AvineshTripathi) [inline providers/azure/networking/local_network_gateways.go:66]: can we fix the failure?

Comment (azgcloudev) [reply providers/azure/networking/local_network_gateways.go:66]: fixed

File: providers/azure/resourcegroup/resource_groups.go (hunk 1 above)
Comment (azgcloudev) [reply providers/azure/resourcegroup/resource_groups.go:33]: I made the change to add the resource as a model and not the string and fill the details.

However I do not added to to listOfSupportedServices as a resource group is not a provider.

**Review by AvineshTripathi — APPROVED** [review]

File: dashboard/utils/serviceHelper.ts [hunk 3]
```diff
@@ -87,7 +87,8 @@ export const allProvidersServices: { [key in Providers]: string[] } = {
     'firewall',
     'load balancer',
     'databox',
-    'queue'
+    'queue',
+    'Local Network Gateway'
```
Comment (Azanul) [inline dashboard/utils/serviceHelper.ts:91]: small case

**Review by Azanul — CHANGES_REQUESTED** [review]
other than this nit, LGTM

File: dashboard/utils/serviceHelper.ts (hunk 3 above)
Comment (azgcloudev) [reply dashboard/utils/serviceHelper.ts]: fixed

**Review by Azanul — APPROVED** [review]

**Pull request merged**